admin/admin.go

```diff
@@ -20,18 +22,27 @@ type AdminServer struct {
 	cancel context.CancelFunc
 	wg     sync.WaitGroup
 
-	networkConfig *config.NetworkConfig
+	networkConfig    *config.NetworkConfig
+	l2ToL2MsgIndexer *interop.L2ToL2MessageIndexer
 
 	port uint64
 }
 
 type RPCMethods struct {
-	Log           log.Logger
-	NetworkConfig *config.NetworkConfig
+	log              log.Logger
+	networkConfig    *config.NetworkConfig
+	l2ToL2MsgIndexer *interop.L2ToL2MessageIndexer
 }
 
-func NewAdminServer(log log.Logger, port uint64, networkConfig *config.NetworkConfig) *AdminServer {
-	return &AdminServer{log: log, port: port, networkConfig: networkConfig}
+func NewAdminServer(log log.Logger, port uint64, networkConfig *config.NetworkConfig, indexer *interop.L2ToL2MessageIndexer) *AdminServer {
+
+	adminServer := &AdminServer{log: log, port: port, networkConfig: networkConfig}
+
+	if networkConfig.InteropEnabled && indexer != nil {
+		adminServer.l2ToL2MsgIndexer = indexer
+	}
```

Comment:
nit- if it's nil then it will be initialized correctly
```
	adminServer := &AdminServer{log: log, port: port, networkConfig: networkConfig, l2ToL2MsgIndexer: indexer}
```